prov/efa/src/rdm/rxr_op_entry.c

```diff
@@ -290,3 +294,632 @@ size_t rxr_tx_entry_num_req(struct rxr_op_entry *tx_entry, int pkt_type)
 	return (total_size - tx_entry->bytes_sent - 1)/tx_entry->max_req_data_size + 1;
 }
 
+/**
+ * @brief handle the situation that an error has happened to an RX (receive) operation
+ *
+ * This function does the following to handle error:
+ *
+ * 1. write an error cq entry to notify application the rx
+ *    operation failed. If write failed, it will write an eq entry.
+ *
+ * 2. increase error counter.
+ *
+ * 3. print warning message about the error with self and peer's
+ *    raw address.
+ *
+ * 4. release resources owned by the RX entry, such as unexpected
+ *    packet entry, because the RX operation is aborted.
+ *
+ * 5. remove the rx_entry from queued rx_entry list for the same reason.
+ *
+ * Note, It will NOT release the rx_entry because it is still possible to receive
+ * packet for this rx_entry.
+ *
+ * @param[in]	rx_entry	rx_entry encoutnered error
+ * @param[in]	err		positive libfabric error code
+ * @param[in]	prov_errno	positive provider specific error code
+ */
+void rxr_rx_entry_handle_error(struct rxr_op_entry *rx_entry, int err, int prov_errno)
+{
+	struct rxr_ep *ep;
+	struct fi_cq_err_entry err_entry;
+	struct util_cq *util_cq;
+	struct dlist_entry *tmp;
+	struct rxr_pkt_entry *pkt_entry;
+	int write_cq_err;
```

Comment:
Same as below. No need to declare a single-use variable.

Reply:
help readability